doc/api/diagnostics_channel.md

```diff
@@ -0,0 +1,189 @@
+diagnostics_channel.subscribe((name, data) => {
+  // Received data
+});
+
+// Get a reusable channel object
+const channel = diagnostics_channel.publisher('my-channel');
+
+// Subscribe directly to a channel object
+channel.subscribe((name, data) => {
+  // Received data
+});
+
+// Check if the channel has an active subscriber
+if (channel.shouldPublish()) {
+  // Publish data to the channel
+  channel.publish({
+    some: 'data'
+  });
+}
+
+// An alternative way to check for active subsribers before publishing
+channel.ifActive?.publish('my-channel', {
+  some: 'data'
+});
+```
+
+#### `diagnostics_channel.subscribe([name, ] onMessage)`
+
+* `name` {string} The channel name
+* `onMessage` {Function} The handler to receive channel messages
+
+Register a message handler to subscribe to existing channels and any future
+channels. The message handler can be a function or an object which includes
+an `onMessage` function. In object form, a `shouldListen` function may also
+be optionally included to filter which channels to subscribe to by running
+the function on the name of the channel whenever a new channel is created.
+
+```js
+const diagnostics_channel = require('diagnostics_channel');
+
+// Subscribe to a channel by name
+diagnostics_channel.subscribe('my-channel', (name, data) => {
+  // Received data
+});
+
+// Subscribe to any channels matching a function result
+diagnostics_channel.subscribe({
+  shouldListen(name) {
+    return /^\s+-channel/.test(name);
+  },
+  onMessage(name, data) {
+    // Received data
+  }
+});
+
+// Subscribe to every channel
+diagnostics_channel.subscribe((name, data) => {
+  // Received data
+});
+```
+
+#### `diagnostics_channel.publish(name, data)`
+
+* `name` {string} The channel name
+* `data` {Object} The message to send to the channel
+
+Lookup a channel by name and send a message, if it has active subscribers.
+It is recommended to use channel objects directly, in most cases, rather than
+using this top-level publish. This exists mainly for convenience in situations
+where publish performance is not important.
+
+```js
+const diagnostics_channel = require('diagnostics_channel');
+
+diagnostics_channel.publish('my-channel', {
+  some: 'data'
+});
+```
+
+#### `diagnostics_channel.publisher(name)`
+
+* `name` {string} The channel name
+* Returns: {Publisher} The named channel object
+
+This is the primary entry-point to anyone wanting to publish data to a named
+channel. It produces a channel object which is optimized to reduce overhead at
+publish time as much as possible.
+
+```js
+const diagnostics_channel = require('diagnostics_channel');
+
+const channel = diagnostics_channel.publisher('my-channel');
+```
+
+### Class: `Publisher`
+
+The class `Publisher` represents an individual named channel within the data
+pipeline. It is primarily used by the publishing side of the event system to
+check if there are subscribers and to send messages.
+
+#### `publisher.shouldListen()`
```

Comment:
Seems like an odd name. hasSubscribers perhaps?

Reply:
Yep, as said above: this is inherited from the original module Microsoft wrote.